modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaSourceModelParser.java

```diff
@@ -192,8 +194,10 @@ private Optional<Method> extractJavaMethod(Optional<MethodElement> method) {
   }
 
   @Override
-  public List<String> getEmittedNotifications() {
-    return NotificationModelParserUtils.getEmittedNotifications(sourceElement, getComponentTypeName(), getName());
+  public List<NotificationModel> getEmittedNotifications(Function<String, Optional<NotificationModel>> notificationMapper) {
+    List<String> identifiers =
+        NotificationModelParserUtils.getEmittedNotifications(sourceElement, getComponentTypeName(), getName());
```

Comment:
avoid unnecesary iterations and collections by having `NotificationModelParserUtils.getEmittedNotifications` delegate into an extracted `getEmmitedNotificationsStream` method.

Reply:
done